File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityEnumTest.cpp
#include "ActivityEnumTest.h"
#include "EnumTrafficLightColor.h"
#include <alf/library/BasicInputOutput.hpp>
#include "EnumTrafficLightColorToString.h"
#include <string>

void EnumTest()
{
  EnumTrafficLightColor enumValue = EnumTrafficLightColor::GREEN;

  alf::library::BasicInputOutput::WriteLine(EnumTrafficLightColorToString(EnumTrafficLightColor::GREEN));

  if (enumValue == EnumTrafficLightColor::GREEN)
  {
    alf::library::BasicInputOutput::WriteLine(std::string("GREEN!"));
  }

  enumValue = EnumTrafficLightColor::RED;

  if (enumValue == EnumTrafficLightColor::GREEN)
  {
    alf::library::BasicInputOutput::WriteLine(std::string("GREEN!"));
  }
  else if (enumValue == EnumTrafficLightColor::RED)
  {
    alf::library::BasicInputOutput::WriteLine(std::string("RED!"));
  }
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityEnumTest.cs
/// <filename>
///     ActivityEnumTest.cs
/// </filename>
using Alf.Library;

public class ActivityEnumTest
{
    public static void Execute()
    {
        EnumTrafficLightColor enumValue = EnumTrafficLightColor.GREEN;

        BasicInputOutput.WriteLine(EnumTrafficLightColorToString.Execute(EnumTrafficLightColor.GREEN));

        if (enumValue == EnumTrafficLightColor.GREEN)
        {
            BasicInputOutput.WriteLine("GREEN!");
        }

        enumValue = EnumTrafficLightColor.RED;

        if (enumValue == EnumTrafficLightColor.GREEN)
        {
            BasicInputOutput.WriteLine("GREEN!");
        }
        else if (enumValue == EnumTrafficLightColor.RED)
        {
            BasicInputOutput.WriteLine("RED!");
        }
    }

    public static void Main(string[] args)
    {
        ActivityEnumTest.Execute();
    }
}
